File: lab2 and lab3/FragmentTabApp/app/src/main/java/kz/mobile/fragmentapp/MainActivity.java
package kz.mobile.fragmentapp;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity implements FirstFragment.SendMessage {


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.d("lifecycle", "Main activity- onRestoreInstanceState" );

        //Show fragment in container
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.fragmentContainer1, new FirstFragment())
                .commit();

        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.fragmentContainer2, new SecondFragment())
                .commit();
    }

    @Override
    public void sendData(String counter) {
        String tag = "android:switcher:" + R.id.viewPager + ":" + 1;
        Log.d("lifecycle", "Main activity send Data");
        SecondFragment fragment = (SecondFragment)
                getSupportFragmentManager().findFragmentByTag(tag);
        fragment.displayReceivedData(counter);
    }
}

File: kotlin example/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {

    private var recyclerView:RecyclerView?=null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        recyclerView=findViewById(R.id.recycleView)


    }



}
